src/components/biggive-tabbed-content/biggive-tabbed-content.tsx

```diff
@@ -1,24 +1,125 @@
-import { Component, Prop, h } from '@stencil/core';
+import { Component, Prop, Element, h } from '@stencil/core';
+import { brandColour } from '../../globals/brand-colour';
+import { spacingOption } from '../../globals/spacing-option';
 
 @Component({
   tag: 'biggive-tabbed-content',
   styleUrl: 'biggive-tabbed-content.scss',
   shadow: true,
 })
 export class BiggiveTabbedContent {
-  /**
-   * Space below component
-   */
-  @Prop() spaceBelow: number = 0;
+  @Element() host: HTMLBiggiveTabbedContentElement;
 
-  /**
-   * Default text colour
-   */
-  @Prop() defaultTextColour: string = 'black';
+  @Prop() spaceBelow: spacingOption = 0;
+
+  @Prop() textColour: brandColour = 'black';
+
+  @Prop() selectedTextColour: brandColour = 'primary';
+
+  @Prop() navigationHighlightColour: brandColour = 'grey-medium';
+
+  @Prop() selectedNavigationHighlightColour: brandColour = 'primary';
+
+  @Prop() buttonBackgroundColour: brandColour = 'white';
+
+  @Prop() buttonIconColour: brandColour = 'primary';
+
+  currentTab = 0;
+  children: Array<any> = [];
+
+  componentWillLoad() {
+    this.children = Array.from(this.host.children);
+  }
+
+  componentDidRender() {
+    this.showTab(0);
+  }
+
+  showTab(i: number) {
+    let tabs = this.host.shadowRoot?.querySelectorAll('.navigation ul li');
+
+    if (tabs) {
+      if (i < 0) {
+        i = 0;
+      } else if (i > tabs?.length - 1) {
+        i = tabs?.length - 1;
+      }
```

Comment:
What's the reason for clamping i into the range of `0` to `tabs?.length - 1`?

Reply:
I have updated so that the next/previous is only triggered if the i is within range